taskomatic/jobs/notificator.py

```diff
@@ -0,0 +1,246 @@
+             .join(InsightsRule, on=(CveRuleMapping.rule_id == InsightsRule.id))
+             .where(InsightsRule.active == True)
+             .where(SystemVulnerabilities.cve_id << notifiable_cves)
+             .where(NotifiedAccounts.notif_type.is_null(True))).dicts()
+    for cve_customer in query:
+        context = make_notification_context(cve_customer)
+        send_kafka_notification(cve_customer["account_id"], NotificationType.RULE_NOTIFICATION, context, loop=MAIN_LOOP)
+        notified_accounts.append((cve_customer["rh_account_id"], cve_customer["cve_id"], NotificationType.RULE_NOTIFICATION))
+    return notified_accounts
+
+
+def notificator_exploit(notifiable_cves: []):
+    """Evaluates if one of notifiable cves has Known Exploits and notifies account"""
+    # pylint: disable=singleton-comparison
+    notified_accounts = []
+    query = make_base_cve_query()
+    query = (query
+             .join(NotifiedAccounts, JOIN.LEFT_OUTER, on=((SystemPlatform.rh_account_id == NotifiedAccounts.rh_account_id) &
+                                                          (CveMetadata.id == NotifiedAccounts.cve_id) &
+                                                          (NotifiedAccounts.notif_type == NotificationType.EXPLOITS_NOTIFICATION)))
+             .where(CveMetadata.exploits == True)
+             .where(SystemVulnerabilities.cve_id << notifiable_cves)
+             .where(NotifiedAccounts.cve_id.is_null(True))).dicts()
+    for cve_customer in query:
+        context = make_notification_context(cve_customer)
+        send_kafka_notification(cve_customer["account_id"], NotificationType.EXPLOITS_NOTIFICATION, context, loop=MAIN_LOOP)
+        notified_accounts.append((cve_customer["rh_account_id"], cve_customer["cve_id"], NotificationType.EXPLOITS_NOTIFICATION))
+    return notified_accounts
+
+
+def retrieve_notified_accounts(futures: []):
+    """Call notificator for each notifiable cve, and notify if condition holds"""
+    notified_accounts = []
+    for future in futures:
+        notificator_notified_accs = future.result()
+        notified_accounts += notificator_notified_accs
+    return notified_accounts
+
+
+def update_notified_accounts(notified_accounts: []):
+    """Update notified accounts in table"""
+    LOGGER.info("Updating %s notified CVE-account pairs", len(notified_accounts))
+    (NotifiedAccounts.insert_many(notified_accounts)
+                     .on_conflict_ignore()
+                     .execute())
+
+
+def clear_notified_accounts(cve_deadline: datetime):
+    """Clear notified accounts for CVEs that are not fresh anymore"""
+    LOGGER.info("Clearing old notified CVE-account notifcation pairs")
+    (NotifiedAccounts.delete()
+                     .where(NotifiedAccounts.cve_id << (
+                         CveMetadata.select(CveMetadata.id)
+                         .where(CveMetadata.modified_date < cve_deadline)
```

Comment:
Old CVEs with exploits will be deleted and on the next run selected again in `query_notifiable_cves` and notified again.